packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -177,6 +193,7 @@ export class EngineReportingExtension<TContext = any>
     // isn't actually in the document. We want to know the name in that case
     // too, which is why it's important that we save the name now, and not just
     // rely on requestContext.operationName (which will be null in this case).
+    this.gqlValidationSucceeded = true;
```

Comment:
Similarly, at least have a comment that this would better belong in validationDidEnd. Though frankly you could also just change graphql-extensions to pass validationErrors to validationDidEnd. I made plenty of graphql-extensions changes as needed by this file.

Reply:
It seemed like making more extensions-specific changes would make this logic harder to port to the new plugin API in a way that was pretty certainly in alignment.

Reply:
That would actually align the stack API with the new plugin API, which already passes validation errors to its version of validationDidEnd.

Reply:
Also `graphql-extensions` is a bit of a beast I just prefer not to touch 😅 Along with the fact that we already know there is overhead for each function call we're putting onto the stack, which is currently causing aer to spew tons of garbage, adding another function for `validationDidEnd` is not something I really really want to do, but if you want to push for it, I shall comply :P

Reply:
The function already exists, it just doesn't receive the list of errors...